src/components/mdTable/mdTablePagination.vue

```diff
@@ -45,26 +45,46 @@
     },
     data() {
       return {
-        subTotal: 0,
         currentSize: parseInt(this.mdSize, 10),
         currentPage: parseInt(this.mdPage, 10),
-        totalItems: isNaN(this.mdTotal) ? Number.MAX_SAFE_INTEGER : parseInt(this.mdTotal, 10)
+        totalItems: parseInt(this.mdTotal, 10)
       };
     },
     computed: {
+      from() {
+        return (this.currentPage - 1) * this.currentSize + 1;
+      },
+      to() {
+        return this.currentPage * this.currentSize;
+      },
+      toCorrected() {
+        return this.isLastPage ? this.mdTotal : this.to;
+      },
+      total() {
+        return isNaN(this.mdTotal) ? Number.MAX_SAFE_INTEGER : this.mdTotal;
+      },
       lastPage() {
-        return false;
+        return Math.ceil(this.totalItems / this.currentSize);
+      },
+      isLastPage() {
+        return this.to >= this.mdTotal;
+      }
+    },
+    watch: {
+      lastPage: function() {
+        if (this.isLastPage) {
+          this.currentPage = this.lastPage;
+        }
       }
     },
     methods: {
       emitPaginationEvent() {
         if (this.canFireEvents) {
-          const sub = this.currentPage * this.currentSize;
-
-          this.subTotal = sub > this.mdTotal ? this.mdTotal : sub;
-          this.$emit('pagination', {
-            size: this.currentSize,
-            page: this.currentPage
+          this.$nextTick(() => {
```

Comment:
yes, it was to ensure currentPage get updated after changing pageSize.
![3](https://cloud.githubusercontent.com/assets/1898273/21798599/4d00debe-d750-11e6-9f77-dd23e71c78b6.png)
![1](https://cloud.githubusercontent.com/assets/1898273/21798614/6844afd4-d750-11e6-8698-970469e3846d.png)